ui/components/src/ui-components/table/component/Table.tsx

```diff
@@ -385,10 +385,13 @@ export class Table extends React.Component<TableProps, TableState> {
   }
 
   /** @internal */
-  public componentDidUpdate(previousProps: TableProps) {
+  public componentDidUpdate(previousProps: TableProps, previousState: TableState) {
     this._rowSelectionHandler.selectionMode = this.props.selectionMode ? this.props.selectionMode : SelectionMode.Single;
     this._cellSelectionHandler.selectionMode = this.props.selectionMode ? this.props.selectionMode : SelectionMode.Single;
 
+    if (previousState.hiddenColumns.length !== this.state.hiddenColumns.length)
```

Comment:
I'd move this down under data provider checks. Also, please add a comment explaining why `forceUpdate` is needed

Reply:
Fixed